scripts/zones/VeLugannon_Palace/npcs/qm3.lua

```diff
@@ -0,0 +1,32 @@
+-----------------------------------
+-- Area: VeLugannon Palace
+--  NPC: qm3 (???) 17502583
+-- Note: Involved in Bartholomew's Knife mini-quest
+-- !pos 0.21 0.57 -322.4 177
+-----------------------------------
+local ID = require("scripts/zones/VeLugannon_Palace/IDs")
+require("scripts/globals/npc_util")
+-----------------------------------
+
+function onTrade(player, npc, trade)
+    if npcUtil.tradeHas(trade, 17622) then -- Buccaneer's Knife
+        if npc:getLocalVar("PillarCharged") == 1 and math.random(1000) <= 50 then -- 50/1000 chance to obtain on trade.
+            player:confirmTrade()
+            player:messageSpecial(ID.text.KNIFE_CHANGES_SHAPE, 17622)
+            npcUtil.giveItem(player, 17623) -- Btm. Knife
+        else
+            player:messageSpecial(ID.text.NOTHING_HAPPENS)
```

Comment:
My understanding is, if you trade that knife and don't get it, it essentially "breaks" like an orb and you can never trade that same knife again. You need to steal a fresh one.

This may require a core change unless you can use the same mechanic we use to break orbs?

Reply:
Testimonies clearly conflict on this.  That aside, burning these knives somehow will require some special code if that's how you guys want to do it.

Reply:
Well... it would be nice to implement this once and completely, but that's not my call to make. The "good enough" approach bugs me because we'll never come back to it, but again that's personal preference.

Reply:
I looked at the main page for the knife again, someone said in 2020 that it is guaranteed burnt when you try to trade and it fails... Implementation of this is beyond my knowledge, but don't let this PR languish just because it's not 100% :(

Reply:
After delving into how WS points are stored for weapons, we could use the "extra" field in char_inventory for each item to store if it's burnt or not.  The exact usage of this field is a little beyond me, but with a little guidance I could add it if you guys want it.

Reply:
> Well... it would be nice to implement this once and completely, but that's not my call to make. The "good enough" approach bugs me because we'll never come back to it, but again that's personal preference.

if someone can _prove it_ I'll make sure it isn't forgotten. I don't trust some random comment on wiki to be the end all be all for it though. Whichever way its implemented initially, this is still worthy of an issue until verified.